specification/hybridconnectivity/resource-manager/Microsoft.HybridConnectivity/stable/2023-03-15/hybridconnectivity.json

```diff
@@ -0,0 +1,1093 @@
+      "description": "Endpoint details",
+      "required": [
+        "type"
+      ],
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "The type of endpoint.",
+          "enum": [
+            "default",
+            "custom"
+          ],
+          "x-ms-enum": {
+            "name": "type",
+            "modelAsString": true
+          }
+        },
+        "resourceId": {
+          "type": "string",
+          "description": "The resource Id of the connectivity endpoint (optional)."
+        },
+        "provisioningState": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The resource provisioning state."
+        }
+      }
+    },
+    "ServiceConfigurationList": {
+      "description": "The paginated list of serviceConfigurations",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The list of service configuration",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ServiceConfigurationResource"
+          }
+        },
+        "nextLink": {
+          "description": "The link to fetch the next page of connected cluster",
+          "type": "string"
+        }
+      }
+    },
+    "EndpointResource": {
+      "description": "The endpoint for the target resource.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "systemData": {
+          "description": "System data of endpoint resource",
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData",
+          "x-ms-client-flatten": true
+        },
+        "properties": {
+          "description": "The endpoint properties.",
+          "$ref": "#/definitions/EndpointProperties",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "ServiceConfigurationResource": {
+      "description": "The service configuration details associated with the target resource.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "systemData": {
+          "description": "System data of endpoint resource",
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData",
+          "x-ms-client-flatten": true
+        },
+        "properties": {
+          "description": "The service configuration properties.",
+          "$ref": "#/definitions/ServiceConfigurationProperties",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "ServiceConfigurationProperties": {
+      "type": "object",
+      "description": "Service configuration details",
+      "required": [
+        "serviceName"
```

Comment:
>serviceName [](http://example.com/codeflow?start=9&length=11)

In most other APIs serviceName is not required. Any reason not to be consistent here?

Reply:
This is used by the API involving create operations, so we do require serviceName param here.
In the other APIs, it is optional if this needs to be accessed or not.